src/pipeline/pipeline_construction.c

```diff
@@ -140,6 +140,53 @@ static ResultProcessor *getAdditionalMetricsRP(RedisSearchCtx* sctx, const Query
   return RPMetricsLoader_New();
 }
 
+static bool AggregateConsumeAll(CommonPipelineParams *cpp, PLN_ArrangeStep *astp) {
+  bool unlimited = false;
+
+  if (IsAggregate(cpp))
+    if (HasDepleter(cpp)) {
+    // FT.AGGREGATE + depleter
+    unlimited = true;
+    } else if (!HasDepleter(cpp) && HasWithCount(cpp) && HasSortBy(cpp) &&
```

Comment:
Is the sortby logic redundant here since also done in AREQ_Compile? (can rely only on `HasDepleter`)

Reply:
Removed.